File: kyonggiyo-core/src/main/java/kyonggiyo/application/service/user/UserQueryService.java
package kyonggiyo.application.service.user;

import kyonggiyo.application.port.in.user.ValidateNicknameUseCase;
import kyonggiyo.application.port.out.user.ExistUserPort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class UserQueryService implements ValidateNicknameUseCase {

    private final ExistUserPort existUserPort;

    @Override
    public boolean existByNickname(String nickname) {
        return existUserPort.existByNickname(nickname);
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/service/user/UserQueryService.kt
package kyonggiyo.application.service.user

import kyonggiyo.application.port.`in`.user.ValidateNicknameUseCase
import kyonggiyo.application.port.out.user.ExistUserPort
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserQueryService(
    private val existUserPort: ExistUserPort
): ValidateNicknameUseCase {

    override fun existByNickname(nickname: String): Boolean =
        existUserPort.existByNickname(nickname)

}
